forem/forem
# Better error upon Tweet embed when Twitter keys missing

## What type of PR is this? (check all applicable)


## Description
A Forem member [noted](https://forem-team.slack.com/archives/CAA200SH5/p1647569715118609?thread_ts=1647567730.787649&cid=CAA200SH5) that, sometimes, tweet embeds fail with the error message `base: Bad Authentication data`. This happens when the Twitter key and secret are missing from `Admin > Customization > Config`.
This PR improves that error message, to inform the user of how to potentially solve the problem.

## QA Instructions, Screenshots, Recordings
First, ensure that there is no Twitter key and secret set in `/admin`.
Next, for the tweet id that you've chosen, check first that the tweet has not been saved in the database: in Rails console, run `Tweet.find_by(tweet_id_code: <tweet-id>)` and ensure it returns `nil`.
Then, create an embed from a Tweet. The embed should fail with the error message: `Authentication error; check that Twitter key and secret are present in Admin`.

### UI accessibility concerns?
None

## Added/updated tests?


## [Forem core team only] How will this change be communicated?
